modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/CatalogUtils.java

```diff
@@ -155,9 +204,10 @@ public static CatalogZoneDescriptor fromParams(int id, CreateZoneParams params)
      * @return Column descriptor.
      */
     public static CatalogTableColumnDescriptor fromParams(ColumnParams params) {
-        int precision = params.precision() != null ? params.precision() : 0;
-        int scale = params.scale() != null ? params.scale() : 0;
-        int length = params.length() != null ? params.length() : 0;
+        int precision = params.precision() != null ? params.precision() : defaultPrecision(params.type());
```

Comment:
Maybe use **java.util.Objects#requireNonNullElse** ? =)

Reply:
1. java.util.Objects#requireNonNullElse will force argument evaluation.

Reply:
At your discretion, this operation does not seem to be performed often, and you can omit this nuance.